connection.js

```diff
@@ -343,32 +343,34 @@ class Connection {
     // Check challenge booleans allow a game ("nopause" is in game.js, not here)
     //
     checkChallengeBooleans(notification) {
-
-        if (config.proonly && !notification.user.professional) {
-            conn_log(`${notification.user.username} is not a professional`);
-            const msg = "You are not a professional player, this bot "
-                        + "accepts games vs professionals only. ";
-            return { reject: true, msg };
-        }
-        if (config.rankedonly && !notification.ranked) {
-            conn_log("Ranked games only");
-            const msg = "This bot accepts ranked games only. ";
-            return { reject: true, msg };
-        }
-        if (config.unrankedonly && notification.ranked) {
-            conn_log("Unranked games only");
-            const msg = "This bot accepts Unranked games only. ";
-            return { reject: true, msg };
-        }
-        if (notification.handicap === -1) {
-            if (config.noautohandicap && !config.noautohandicapranked && !config.noautohandicapunranked) {
-                return noAutohandicapReject("noautohandicap");
-            }
-            if (config.noautohandicapranked && notification.ranked) {
-                return noAutohandicapReject("noautohandicapranked");
+        const testBooleanArgs     = [ ["rankedonly", "Unranked games are", !notification.ranked],
```

Comment:
Instead of making a for loop with a fixed list. You should make the content of the for loop a funciton call, and call it with 
```Js
testBooleanArgs("rankedonly", "Unranked games are", !notification.ranked);
testBooleanArgs("unrankedonly", "Ranked games are", notification.ranked);
```

Reply:
In fact if i read this code right, the for loops does nothing for one of them, so you could call:
```
if (notification.ranked) { 
  testBooleanArgs("unrankedonly", "Ranked games are");
} else
  testBooleanArgs("rankedonly", "Unranked games are");
}
```
and remove the `notifCondition` variable.

Reply:
He's going to use this format more generally in future PRs I believe, so they all work the same way.

Reply:
Yes that's why he should start using the correct format now :-)